From 65ea8d19d91ac1599a29e8411485a72fe89c45e2 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.xyz>
Date: Wed, 28 Dec 2022 17:58:47 +0100
Subject: [PATCH] vout_opengl: add compatibility alias for libplacebo v5

This fixes compilation for now, albeit with many deprecation warnings.
It will probably break again whenever libplacebo v6 drops these
deprecated fields as well, but hopefully VLC v4 will be released by
then.

Fixes: #27624
---
 modules/video_output/opengl/vout_helper.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/video_output/opengl/vout_helper.h b/modules/video_output/opengl/vout_helper.h
index db9ed52e395..48014b454c4 100644
--- a/modules/video_output/opengl/vout_helper.h
+++ b/modules/video_output/opengl/vout_helper.h
@@ -34,6 +34,11 @@
 #ifdef HAVE_LIBPLACEBO
 #include <libplacebo/shaders/colorspace.h>
 
+# if PL_MAJOR_VER >= 5
+# define pl_context_create pl_log_create
+# define pl_context_destroy pl_log_destroy
+# endif
+
 #define RENDER_INTENT_TEXT "Rendering intent for color conversion"
 #define RENDER_INTENT_LONGTEXT "The algorithm used to convert between color spaces"
 
-- 
GitLab

